src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -377,16 +388,17 @@ internal DefinitionResult GetDefinitionErrorResult(string errorMessage)
         /// </summary>
         /// <param name="quickInfoText">QuickInfo Text for this token</param>
         /// <param name="tokenText">Token Text</param>
+        /// <param name="caseSensitivity">StringComparison enum</param>
         /// <returns></returns>
-        internal string GetFullObjectNameFromQuickInfo(string quickInfoText, string tokenText)
+        internal string GetFullObjectNameFromQuickInfo(string quickInfoText, string tokenText, StringComparison caseSensitivity)
         {
             if (string.IsNullOrEmpty(quickInfoText) || string.IsNullOrEmpty(tokenText))
             {
                 return null;
             }
             // extract full object name from quickInfo text
             string[] tokens = quickInfoText.Split(' ');
-            List<string> tokenList = tokens.Where(el => el.Contains(tokenText)).ToList();
+            List<string> tokenList = tokens.Where(el => el.IndexOf(tokenText, caseSensitivity) >= 0).ToList();
```

Comment:
Using IndexOf seems more straightforward and we can keep the case sensitivity check's code pattern consistent everywhere.